### PR TITLE
Regapp

### PR DESCRIPTION
**Pull request by vviers**
Hey!

I'm done with the regression app. Few comments:

1. I put the app in a standalone .rmd (that can be copied and pasted into the chapter once it's finalized)

1. The app generates 10 examples (this is arbitrary, can theoretically generate an infinity of examples)
1. I chose to not only show the error lines but also, in light red (or green in the right slope is found), the boxes corresponding to the **squared** errors. I really think it helps to understand what is going on!
1. I also report MSE and the correlation coefficient.
1.  lastly, there is a slider that allows to select the noisiness of the generated data

Lmk what you think

V

### CODE REVIEW

**Comment by floswald** [comment]
so it seems all the commits from the previous PR are included here again. not sure why that happened. you probably branched off the `distribs` branch instead of a clean master? not sure.

File: inst/tutorials/chapter4/reg_app_standalone.Rmd [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  expr <- function(x) a + b*x
+  errors <- (a + b*x) - y 
+  
+  plot(x, y, type = "p", pch = 21, col = "blue", bg = "royalblue", asp=1,
+       xlim = c(min(c(x, y))-1, max(c(x, y))+1),
+       ylim = c(min(c(x, y))-1, max(c(x, y))+1),
+       main = "The Linear Regression", frame.plot = FALSE,
+       cex = 1.2)
+  
+  legend("topleft", legend = paste0("r = ", round(cor(x, y), 3)))
```
Comment (floswald) [inline inst/tutorials/chapter4/reg_app_standalone.Rmd]: explain what r is

File: inst/tutorials/chapter4/reg_app_standalone.Rmd [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  if (a == a_true && b == b_true){
+    curve(expr = expr, from = min(x)-10, to = max(x)+10, add = TRUE, col = "black")
+    segments(x0 = x, y0 = y, x1 = x, y1 = (y + errors), col = "green")
+    rect(xleft = x, ybottom = y,
+         xright = x + abs(errors), ytop = y + errors, density = -1,
+         col = rgb(red = 0, green = 1, blue = 0, alpha = 0.1), border = NA)
+  } else {
+    curve(expr =expr , from = min(x)-10, to = max(x)+10, add = TRUE, col = "black")
+    segments(x0 = x, y0 = y, x1 = x, y1 = (y + errors), col = "red")
+    rect(xleft = x, ybottom = y,
```
Comment (floswald) [inline inst/tutorials/chapter4/reg_app_standalone.Rmd]: the rectangles are great!

File: inst/tutorials/chapter4/reg_app_standalone.Rmd [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+```{r, echo=FALSE}
+shinyUI(fluidPage(
+  sidebarPanel(sliderInput("i", "Intercept", min = -2,
+                           max = 2, step = .5, value = 0),
+               sliderInput("s", "Slope", min = -2,
```
Comment (floswald) [inline inst/tutorials/chapter4/reg_app_standalone.Rmd]: starting value always not equal zero!

**Comment by vviers** [comment]
I can copy-paste it into the chapter 4 if that's the question, when working on 'standalone' versions of apps I'll always aim at them being fully integrable into the actual book 😄

File: inst/tutorials/chapter4/reg_app_standalone.Rmd [hunk 4]
```diff
@@ -192,10 +192,10 @@ What if you could only use one fixed number (i.e. a horizontal line) to fit $Y$
...
   sidebarPanel(
     
-    radioButtons("const", label = h4("Fixed Parameter?"),
-    choices = list("Slope = 0" = 1,
-                   "Intercept = 0" = 2,
-                   "No" = 3), 
+    radioButtons("const", label = h4("Parameter"),
+    choices = list("Intercept only" = 1,
+                   "Slope only" = 2,
+                   "Both" = 3),
```
Comment (floswald) [inline inst/tutorials/chapter4/reg_app_standalone.Rmd]: not "both" but "standard"

**Pull request merged**